chain/client/src/client_actor.rs

```diff
@@ -387,7 +387,9 @@ impl Handler<NetworkClientMessages> for ClientActor {
                                 }
                                 return NetworkClientResponses::NoResponse;
                             } else if block.hash() == sync_hash {
-                                if let Err(e) = self.client.chain.save_orphan(block) {
+                                // This is the immediate block after a state sync
+                                // We can afford to delay requesting missing chunks for this one block
+                                if let Err(e) = self.client.chain.save_orphan(block, false) {
```

Comment:
Thanks for adding comments here! I also want to note that this is a ridiculous hackery and there is no guarantee that the block is still in the orphan pool when the chunks arrive